src/dbnode/persist/fs/retriever.go

```diff
@@ -299,12 +306,19 @@ func (r *blockRetriever) fetchBatch(
 
 	// Sort the requests by offset into the file before seeking
 	// to ensure all seeks are in ascending order
+	var entrySize int
 	for _, req := range reqs {
+		if err := r.opts.QueryStats().UpdateBytesRead(entrySize); err != nil {
+			req.onError(err)
+			continue
```

Comment:
Should this break? Don't think we want to keep processing a partial batch here otherwise

Reply:
I think we still want to "complete" each request because internally they have a waitGroup that should get marked. And so that is why we want to set the `onError` for all remaining reqs like this.

Reply:
Would it be worth short circuiting here then? Although I suppose we'd miss recording the cost of the entire query, but it would alleviate pressure on the seeker

```golang
var exceededErr err
for _, req := range reqs {		for _, req := range reqs {
	if exceededErr != nil {
		req.onError(exceededErr)
	if exceededErr = r.opts.QueryStats().UpdateBytesRead(entrySize); err != nil {
		req.onError(exceededErr)
	}
	// ...
```